R/spat4cast_submit.R

```diff
@@ -1,8 +1,22 @@
 
-spat4cast_submit <- function(dir, theme = "lai_recovery") {
-  # upload anonymously to submissions bucket
+spat4cast_submit <- function(dir, theme = "lai_recovery", fire = "august_complex") {
+  
+  submitted_files <- list.files(dir, full = TRUE)
+  
+  # Check file type
+  assertthat::assert_that(n_distinct(tools::file_ext(submitted_files)) == 1)
+  assertthat::assert_that(unique(tools::file_ext(submitted_files)) == "tif", msg = "Error: The file extension in the directory is not 'tif'. Please check the file format.")
+  
+  # Check bounding box for files
+  ## can we use the `fire` argument to pull a template target file for comparison?
+  ## if we are to enforce these attributes, then we should modify `ingest_planetary_data` to remove dy, dx, dt, srs arguments (and have them enforced according to the fire of intered)
+  res(rast(submitted_files))
+  origin(rast(submitted_files))
+  ext(rast(submitted_files))
+
+  ## Upload anonymously to submissions bucket
   minioclient::mc_alias_set("efi", "data.ecoforecast.org", "", "")
   minioclient::mc_cp(dir, 
-                     paste0("efi/spat4cast-submissions/", theme),
+                     paste0("efi/spat4cast-submissions/", theme, "/", fire), # should we have separate buckets for each fire?
```

Comment:
good call, that's probably easiest.  paralleling the existing conventions in efi, I think we should refer to theme as the `variable` and the fire location as the `site_id`, and let's express this in hive format, so maybe something like:

```r
glue("efi/spat4cast-submissions/variable={theme}/site_id={fire}")
```

Reply:
oh somewhere in here we need a `model_id={model_id}` as well, since we will have different models (from the same or different teams of people)  forecasting the same site_id / reference_datetime etc.  (I guess we currently have that in the filenames but might help to have it explicitly on the path).

Reply:
how would you suggest validating `model_id`? would we need to provide guidance on how to identify the team and model type?